IframeMediaPause/iframe_media_pausing.md

```diff
@@ -0,0 +1,189 @@
+
+A website can dynamically modify an embedded iframe using JavaScript when reacting to user input. There could be scenarios where the document loaded in an iframe has many resources and initializing it might require a considerable amount of time. Let’s suppose that, when reacting to some type of user input, the embedder decides to not render the iframe by setting its CSS "display" property to "none". Voila! The iframe has vanished while the resources are still loaded. However, what if this iframe’s document was also playing some kind of audible media? The iframe will not be rendered anymore, but the audio keeps playing. 
+
+Some websites, like [YouTube](https://developers.google.com/youtube/iframe_api_reference) and [Vimeo](https://github.com/vimeo/player.js/), offer APIs that the embedding document can call to control the media in a directly embedded iframe – i.e. a "child" iframe. However, it wouldn’t work if the embedding document were trying to control a more distant iframe, like a grandchild iframe; unless the child iframe shares the same origin of the parent iframe and/or also provides an API to communicate with its parent.
+
+## Goals
+
+Propose a mechanism to allow embedder documents to limitedly control embedded iframe media playback based on whether the embedded iframe is [rendered](https://html.spec.whatwg.org/multipage/rendering.html#being-rendered) or not:
+- When the iframe is not rendered, the embedder is able to pause the iframe media playback; and
+-  When the iframe becomes rendered again, the embedder is able to resume the iframe media playback.
+
+## Non-Goals
+It is not a goal of this proposal to allow embedders to arbitrarily control when to play, pause, stop, resume, etc, the media playback of a rendered iframe.
+
+## Use Cases
+There are scenarios where a website might want to just not render an iframe. For example:
+- A website, in response to an user action, might decide to temporarily not show an iframe that is playing media. However, since it is not possible to mute it, the only option is for the website to remove the iframe completely from the DOM and recreate it from scratch when it should be visible again. Since the embedded iframe can also load many resources, the iframe recreation operation might make the web page slow and spend resources unnecessarily.
+
+## Proposed solution: media-playback-while-not-rendered Permission Policy
+
+We could create new "media-playback-while-not-rendered" [Permission Policy](https://www.w3.org/TR/permissions-policy/) that would pause any media being played by iframes which are not currently [rendered](https://html.spec.whatwg.org/multipage/rendering.html#being-rendered). This would apply whenever the iframe’s `"display"` CSS property is set to `"none"`.
+
```

Comment:
"This would apply whenever the iframe’s `"display"` CSS property is set to `"none"`." 

Should we enumerate any other scenarios where the new proposal would pause media playback?

Reply:
As discussed offline, we couldn't think of any other scenarios where the iframe is not rendered and is still alive, besides setting `"display"` to `"none"`.